noaa_coops/station.py

```diff
@@ -477,6 +457,189 @@ def _parse_known_date_formats(self, dt_string: str):
+        bin_num: Optional[int] = None,
+        interval: Optional[Union[str, int]] = None,
+        units: Optional[str] = "metric",
+        time_zone: Optional[str] = "gmt",
+    ):
+        """Check that requested product parameters are valid.
+
+        Args:
+            product (str): Data product to be fetched.
+            datum (str, optional): Datum to use for water level products.
+            bin_num (int, optional): Bin to use for current products. Defaults to None.
+            interval (Union[str, int], optional): Time interval of fetched data.
+                Defaults to None
+            units (str, optional): Units to use for fetched data. Defaults to "metric".
+            time_zone (str, optional): Time zone to use for fetched data.
+                Defaults to "gmt".
+
+        Raises:
+            ValueError: Invalid request parameters were provided.
+        """
+        if product not in [
+            "water_level",
+            "hourly_height",
+            "high_low",
+            "daily_mean",
+            "monthly_mean",
+            "one_minute_water_level",
+            "predictions",
+            "datums",
+            "air_gap",
+            "air_temperature",
+            "water_temperature",
+            "wind",
+            "air_pressure",
+            "conductivity",
+            "visibility",
+            "humidity",
+            "salinity",
+            "currents",
+            "currents_predictions",
+            "ofs_water_level",
+        ]:
+            raise ValueError(
+                f"Invalid product '{product}' provided. See"
+                " https://api.tidesandcurrents.noaa.gov/api/prod/#products "
+                "for list of available products"
+            )
+
+        if product in [
+            "water_level",
+            "hourly_height",
+            "high_low",
+            "daily_mean",
+            "monthly_mean",
+            "one_minute_water_level",
+            "predictions",
+        ]:
+            if datum is None:
+                raise ValueError(
+                    "No datum specified for water level data. See"
+                    " https://api.tidesandcurrents.noaa.gov/api/prod/#datum "
+                    "for list of available datums"
+                )
+            elif datum not in [
+                "CRD",
+                "IGLD",
+                "LWD",
+                "MHHW",
+                "MHW",
+                "MTL",
+                "MSL",
+                "MLW",
+                "MLLW",
+                "NAVD",
+                "STND",
+            ]:
+                raise ValueError(
+                    f"Invalid datum '{datum}' provided. See"
+                    " https://tidesandcurrents.noaa.gov/api/prod/#datum "
+                    "for list of available datums"
+                )
+
+        if product in ["water_level", "hourly_height", "one_minute_water_level"]:
+            if interval is not None:
+                raise ValueError(
```

Comment:
I'm not sure how the API handles the `interval` parameter for `water_level`, but for what it's worth, it _does_ work. Only hourly values are returned by `get_data()` if it is specified, whereas without it 6-minute values are returned. I recommend not raising an error here, perhaps a UserWarning that the interval may not be necessary

Reply:
In fact, I've been testing it a little bit, and I think you definitely want to make sure the combination of `product='water_level', interval='h'` works, because it looks like the `hourly_height` product is not available until _after_ the data has been verified, so there is no way to obtain recent (i.e. within the present month) hourly values without going through the `water_level` product.

Reply:
Thank you for testing! Taking a look

Reply:
Ahhh - this is why the interval param works for `water_level` https://github.com/GClunies/noaa_coops/blob/bf55c78ea10fcf80f2de7906e5470919625ce4a7/noaa_coops/station.py#L1000-L1002

TBH, I haven't used this package for _ages_ as I no longer work as a coastal engineer. BUT, I am starting to think that `noaa_coops` should simply return the requested data in a DataFrame as received by the API with no/minimal transformations. That would: 
- Reduce confusion between the CO-OPS API docs and `noaa_coops`
- Put the onus on the _end user_ to handle things like:
  - Filtering to the desired interval for `water_level` and `predictions`
  - Column renames - `noaa_coops` does this for _some products_. This is fragile and could easily break if their are upstream changes in the API.